stock_request/models/stock_request.py

```diff
@@ -97,7 +98,8 @@ def _get_default_requested_by(self):
         states={'draft': [('readonly', False)]}, readonly=True
     )
     route_id = fields.Many2one(
-        states={'draft': [('readonly', False)]}, readonly=True
+        states={'draft': [('readonly', False)],
+                'submitted': [('readonly', False)]}, readonly=True
```

Comment:
You should add this change for all fields.

Reply:
@etobella The warehouse user is not supposed to change fields in submitted state. Route is an exception.